Memorize/MemoryGame.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  MemoryGame.swift
+//  Memorize
+//
+//  Created by Kedar Dhere on 8/10/22.
+//
+
+import Foundation
+
+struct MemoryGame<CardContent> where CardContent: Equatable {
+
+    // MARK: - Properties
+    struct Card: Identifiable {
+
+        var isFaceUp: Bool = false
+        var isMatched: Bool = false
+        var content: CardContent
+        var isPreviouslySeen: Bool = false
+        var id: Int
+
+    }
+
+    private(set) var cards: [Card]
+//    private var indexOfTheOneAndOnlyFaceUpCard: Int?
+    private var indexOfTheOneAndOnlyFaceUpCard: Card?
+    private(set) var score: Int = 0
+
+    // MARK: - Initilization
+    init(numberOfPairsOfCards: Int, createCardContent: (Int) -> CardContent) {
+        cards = [Card]()
+        for pairIndex in 0..<numberOfPairsOfCards {
+            let content: CardContent = createCardContent(pairIndex)
+            cards.append(Card(content: content, id: pairIndex*2))
+            cards.append(Card(content: content, id: pairIndex * 2 + 1))
+        }
+        cards.shuffle()
+    }
+
+    // MARK: - Functions
+    mutating func choose(_ card: Card) {
+        guard
+            var chosenCard = cards.first(where: {$0.id == card.id}),
+            !chosenCard.isFaceUp,
+            !chosenCard.isMatched
+        else {
+            return
+        }
+        print("Chosen Card Initially \(chosenCard)")
```

Comment:
Is this print statement left in intentionally?  Are you trying to log?

Reply:
The print statement was used for debugging purposes, removed as it is not required.